posthog/api/capture.py

```diff
@@ -871,6 +883,11 @@ def capture_internal(
     # overriding this to deal with hot partitions in specific cases.
     # Setting the partition key to None means using random partitioning.
     candidate_partition_key = f"{token}:{distinct_id}"
+    if event.get("properties", {}).get(COOKIELESS_MODE_FLAG_PROPERTY):
+        # In cookieless mode, the distinct id is meaningless, so we can't use it as the partition key.
+        # Instead, use the IP address as the partition key.
+        candidate_partition_key = f"{token}:{ip}"
```

Comment:
People **could** (rarely) change IPs inside the same session. I assume this is a problem we'll have to bear with because we're using the IP in the hash, but we could try and make sure events from the same person are really running in the same partition.

One possibility is using the website host, but that could cause heavy balancing problems between partitions.
We could also use the browser's timezone, but that's likely too low cardinality.

IP is probably the best one, but we need to remember this might be flaky :)

Reply:
> but we could try and make sure events from the same person are really running in the same partition.

I don't think we gain any benefit from this while the IP address is in the hash though